tests/contrib/hooks/test_sagemaker_hook.py

```diff
@@ -0,0 +1,246 @@
+                                'Type': 'Maximize',
+                                'MetricName': 'test_metric'
+                            },
+                            'ResourceLimits': {
+                                'MaxNumberOfTrainingJobs': 123,
+                                'MaxParallelTrainingJobs': 123
+                            },
+                            'ParameterRanges': {
+                                'IntegerParameterRanges': [
+                                    {
+                                        'Name': 'k',
+                                        'MinValue': '2',
+                                        'MaxValue': '10'
+                                    },
+                                ]
+                            }
+                        },
+                        'TrainingJobDefinition': {
+                            'StaticHyperParameters':
+                                create_training_params['HyperParameters'],
+                            'AlgorithmSpecification':
+                                create_training_params['AlgorithmSpecification'],
+                            'RoleArn': 'string',
+                            'InputDataConfig':
+                                create_training_params['InputDataConfig'],
+                            'OutputDataConfig':
+                                create_training_params['OutputDataConfig'],
+                            'ResourceConfig':
+                                create_training_params['ResourceConfig'],
+                            'StoppingCondition': dict(MaxRuntimeInSeconds=60 * 60)
+                        }
+                        }
+
+db_config = {"Re4sourceConfig": {
+    "InstanceCount": 3,
+    "InstanceType": "ml.c4.4xlarge",
+    "VolumeSizeInGB": 25
+}
+}
+
+
+class TestSageMakerHook(unittest.TestCase):
+
+    def setUp(self):
+        configuration.load_test_config()
+        db.merge_conn(
+            models.Connection(
+                conn_id='sagemaker_test_id',
+                conn_type='sagemaker',
+                login='access_id',
+                password='access_key',
+                extra=json.dumps(db_config)
+            )
+        )
+
+    def test_check_valid_training_input(self):
+        hook = SageMakerHook()
+        for key in create_training_params:
+            if isinstance(create_training_params[key], dict):
+                for sub_key in create_training_params[key]:
+                    temp_params = copy.deepcopy(create_training_params)
+                    del temp_params[key][sub_key]
+                    self.assertRaises(AirflowException,
+                                      hook.check_valid_training_input, temp_params)
+            temp_params = copy.deepcopy(create_training_params)
+            del temp_params[key]
+
+            self.assertRaises(AirflowException,
+                              hook.check_valid_training_input, temp_params)
```

Comment:
Not all keys in the config dict are required. So you should come up with a new logic for testing.